src/vs/workbench/contrib/positronIPyWidgets/browser/positronIPyWidgetsService.ts

```diff
@@ -66,23 +66,39 @@ export class PositronIPyWidgetsService extends Disposable implements IPositronIP
 			this._consoleInstancesByMessageId.has(id);
 	}
 
+	/**
+	 * Map to disposeable stores for each session. Used to preventing memory leaks caused by
+	 * repeatedly attaching to the same session which can happen in the case of the application
+	 * closing before the session ends
+	 */
+	static SessionToDisposablesMap = new Map<ILanguageRuntimeSession, DisposableStore>();
```

Comment:
Any reason this can't be private? IIUC this service is a singleton, so `static` was a bit confusing to me

Reply:
Also, should we override `PositronIPyWidgetsService.dispose` to dispose all remaining sessions in this map to handle the case when the service is disposed before the session?

Reply:
I always forget about singleton-ness. I'll privatize it and make it dispose on close!